### PR TITLE
Scratchpad

### PR DESCRIPTION
**Pull request by minus7**
Should be it for scratchpad. One more thing to cross off of #2.

Implemente both, `move scratchpad` and `scratchpad show`

### CODE REVIEW

File: sway/layout.c [hunk 1]
```diff
@@ -415,3 +420,70 @@ void recursive_resize(swayc_t *container, double amount, enum wlc_resize_edge ed
...
+		// Refocus on the view once its been put back into the layout
+		view->width = view->height = 0;
+		arrange_windows(swayc_active_workspace(), -1, -1);
+	}
+}
+
+void scratchpad_push(swayc_t *view) {
+	add_floating(scratchpad, view);
+	wlc_view_set_mask(view->handle, 2); // invisible mask
+    wlc_view_send_to_back(view->handle);
```
Comment (SirCmpwn) [inline sway/layout.c]: style

**Comment by SirCmpwn** [comment]
Code looks good. Will test soon

**Comment by SirCmpwn** [comment]
Good first pass. However, it doesn't match the i3 functionality. Try this from i3:

2. move scratchpad
3. scratchpad show
4. scratchpad show

The second "show" will return the window to the scratchpad.

File: sway/commands.c [hunk 2]
```diff
@@ -762,6 +754,7 @@ static struct cmd_handler handlers[] = {
 	{ "layout", cmd_layout },
 	{ "log_colors", cmd_log_colors },
 	{ "move", cmd_move},
+	{ "scratchpad", cmd_scratchpad },
```
Comment (SirCmpwn) [inline sway/commands.c]: Forgot the whole "keep alphabetized" thing, huh?

**Comment by SirCmpwn** [comment]
How to break this:

1. Move app to scratchpad
2. `workspace right`

Fix before merge

**Comment by taiyu-len** [comment]
should probably add the command to man page aswell.

File: sway/commands.c (hunk 2 above)
Comment (minus7) [inline sway/commands.c]: First I heard of it :D, fixing.

**Comment by minus7** [comment]
More rewriting in hope of getting it right, it's now more broken than before. But! workspace next_on_output works correctly now :)

I get the feeling like remove_child should hide a window as long as it's not attached to anything. Container/view visibility is really a pain currently.

**Comment by SirCmpwn** [comment]
Let me know when it's ready for testing/review.

**Comment by minus7** [comment]
I'll make a new PR when I got it right.

**Pull request closed without merging**